### PR TITLE
Error messages are displayed in the import modal

### PR DESCRIPTION
**Pull request by falkodev**
Error messages are displayed in the import modal now (requested by ECG).

### CODE REVIEW

**Review by boutell — CHANGES_REQUESTED** [review]
What can these errors be? Is it safe to disclose them to the user? Could they be database errors or are they always some kind of appropriate error about a column name for instance?

**Comment by falkodev** [comment]
> What can these errors be? Is it safe to disclose them to the user? Could they be database errors or are they always some kind of appropriate error about a column name for instance?

@boutell It can be errors like "update:notfound" about a column name but also an error coming from a DB query (I guess a name too long or if only digits allowed). But not sure if it reveals confidential information to the user.

**Comment by boutell** [comment]
This should not simply expose all errors to the user. If there are specific errors you can catch that you know are not reflective of any more "technical" details you can reveal them.

**Comment by falkodev** [comment]
> This should not simply expose all errors to the user. If there are specific errors you can catch that you know are not reflective of any more "technical" details you can reveal them.

The only errors I have been able to produce where "update-notfound" and "Document has neither slug nor title, giving up". Nothing revealing the DB structure to an unauthorized user I think. The whole error message indicating all the files and functions implied in the error process are not shown.

**Comment by boutell** [comment]
We should send only the errors that we know are appropriate to send. I assume we're pursuing this because the client asked to be informed of errors they can do something about, perhaps certain specific errors. We should identify those and choose to send them. We should not send all errors, present and future.

File: index.js [hunk 1]
```diff
@@ -379,18 +379,27 @@ module.exports = {
...
+            : err.includes('update-notfound');
+
           return self.db.update({ _id: job._id }, {
             $inc: {
               errors: 1,
               processed: 1
+            },
+            ...displayError && {
+              $addToSet: {
+                errorMessages: err.message || err
```
Comment (boutell) [inline index.js:401]: This is correct, but is it enough information to help them fix it? In the case of "update-notfound" should we add the value that was given for the update key to the error? The "neither slug nor title" one is probably enough information.

**Review by boutell — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (falkodev) [reply index.js:401]: Here is an example of a "not-found" error:

image

It displays the key that is supposed to be updated but that does not exist yet (the title in my example). It is what is displayed currently in the terminal anyway, so no more or less information that a server admistrator has.

**Review by boutell — APPROVED** [review]

**Pull request merged**